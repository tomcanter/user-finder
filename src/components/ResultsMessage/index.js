// @flow

import React, {
  PropTypes,
} from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';

type Props = {
  pageTotal: number,
  resultsTotal: number,
  searchTerm: string,
};

function ResultsMessage({searchTerm, resultsTotal, pageTotal}: Props) {
  if (!searchTerm || !pageTotal) {return null;}

  return (
    <p className={css(styles.ResultsMessage)}>
      Found {resultsTotal} {resultsTotal === 1 ? 'result' : 'results'} for <b>{searchTerm}</b>
    </p>
  );
}

ResultsMessage.propTypes = {
  searchTerm: PropTypes.string,
  resultsTotal: PropTypes.number,
  pageTotal: PropTypes.number,
};

const styles = StyleSheet.create({
  ResultsMessage: {
    textAlign: 'center',
  },
});

export default ResultsMessage;
