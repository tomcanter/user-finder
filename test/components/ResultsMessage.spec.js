import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';

import ResultsMessage from 'components/ResultsMessage';

describe('Component: ResultsMessage', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const component = renderer.create(
      <ResultsMessage />
    );
    expect(component).toMatchSnapshot();
  });

});
