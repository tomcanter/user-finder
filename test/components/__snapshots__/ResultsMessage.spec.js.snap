exports[`Component: ResultsMessage should handle single results 1`] = `
<p
  className="ResultsMessage_1ntyrs8">
  Found 
  1
   
  result
   for 
  <b>
    test
  </b>
</p>
`;

exports[`Component: ResultsMessage should render 1`] = `
<p
  className="ResultsMessage_1ntyrs8">
  Found 
  50
   
  results
   for 
  <b>
    test
  </b>
</p>
`;
